### PR TITLE
Fix dialog

### PR DESCRIPTION
**Pull request by CarolineLaw**
- getting rid of the discard dialog when adding a new phrase
- nick fixed the adapter so that it update correctly

### CODE REVIEW

File: app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt [hunk 1]
```diff
@@ -275,24 +274,24 @@ class PresetsFragment : BaseFragment() {
...
-        var numPages = 0
+        override fun setItems(items: List<Any>) {
+            super.setItems(items)
 
-        fun setCategories(categories: List<Category>) {
-            with(this.categories) {
+            with(categories) {
                 clear()
-                addAll(categories)
+                items.forEach {
```
Comment (MattCorbett488) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt:286]: Think you can condense this down to `addAll( items.filter { it is Category } )`

File: app/src/main/java/com/willowtree/vocable/settings/EditPresetsFragment.kt [hunk 2]
```diff
@@ -191,37 +190,34 @@ class EditPresetsFragment : BaseFragment() {
...
+
+            with(phrases) {
                 clear()
-                addAll(phrases)
-            }
-            numPages = if (phrases.isEmpty()) {
-                1
-            } else {
-                ceil(phrases.size / maxPhrases.toDouble()).toInt()
+                items.forEach {
```
Comment (MattCorbett488) [inline app/src/main/java/com/willowtree/vocable/settings/EditPresetsFragment.kt:202]: with the calls to super, are you not basically doing this twice?

File: app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt (hunk 1 above)
Comment (MattCorbett488) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt:286]: Also, do you expect some of these types to _not_ be `Category`?

Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt:286]: So `items` could be either a list of `Category` or a list of `Phrase`. So I was having trouble casting the `items` list to the right type in each of the subclasses. Would making it generic work for that? So I could have a `VocableFragmentStateAdapter<Category>` and `setItems` would be `setItems(items: List<Category>)`?

Comment (MattCorbett488) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt:286]: It should yeah

You could do `VocableFragmentStateAdapter<T>(fm: FragmentManager, lifecycle: Lifecycle)`
and then change `setItems` to `open fun setItems(items: List<T>)`

Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt]: Cool, just updated the adapters

File: app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt [hunk 3]
```diff
@@ -275,78 +274,48 @@ class PresetsFragment : BaseFragment() {
...
-            notifyDataSetChanged()
+        VocableFragmentStateAdapter<Phrase>(fm, viewLifecycleOwner.lifecycle) {
 
+        override fun setItems(items: List<Phrase>) {
+            super.setItems(items)
             setPagingButtonsEnabled(phrasesAdapter.numPages > 1)
         }
 
         private fun setPagingButtonsEnabled(enable: Boolean) {
             binding?.let {
```
Comment (jstubing) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt:301]: You can alternatively do a binding?.apply here and then you wouldn't need to use "it" in the body

File: app/src/main/java/com/willowtree/vocable/settings/EditCategoriesFragment.kt [hunk 4]
```diff
@@ -208,6 +195,7 @@ class EditCategoriesFragment : BaseFragment() {
         binding?.let {
```
Comment (jstubing) [inline app/src/main/java/com/willowtree/vocable/settings/EditCategoriesFragment.kt:195]: Same deal here. Using apply can clean this up a bit.

File: app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt (hunk 3 above)
Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt:301]: Good point, we'll address these blocks going forward. There's multiple places we use this style.

**Pull request merged**